Error calculating coverage because of NPE in readClasspathManifest

I'm using the Gradle plugin version 1.5.2 to run pitest in a Junit5 project. I've configured pitest only in one sub-project for now (see the end of the issue for repro).

```
plugins {
    id 'info.solidsoft.pitest' version '1.5.2'
}

pitest {
    //adds dependency to org.pitest:pitest-junit5-plugin and sets "testPlugin" to "junit5"
    junit5PluginVersion = '0.12'
    // ...
}
```

It seems like the tests are running, but something messes up the code coverage calculation. 

```
12:26:06 PM PIT >> INFO : MINION : 12:26:06 PM PIT >> INFO : Found  72 tests

12:26:06 PM PIT >> INFO : MINION : 12:26:06 PM PIT >> SEVERE : Error calculating coverage. Process will exit.
java.lang.NullPointerException
        at org.pitest.util.ManifestUtils.readClasspathManifest(ManifestUtils.java:78)
        at org.pitest.classpath.ClassPath.lambda$addEntriesFromClasspathManife
12:26:06 PM PIT >> INFO : MINION : st$2(ClassPath.java:143)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at org.pitest.classpath.ClassPath.addEntriesFromClasspathManifest(ClassPath.java:143)
        at org.pitest.classpath.ClassPath.getClassPathElementsAsFiles(ClassPath.java:128)
12:26:06 PM PIT >> INFO : MINION :
        at org.pitest.classpath.ClassPath.<init>(ClassPath.java:47)
        at org.pitest.classpath.ClassPathByteArraySource.<init>(ClassPathByteArraySource.java:31)
        at org.pitest.coverage.execute.CoverageMinion.getTestsFromParent(CoverageMinion.java:140)
        at org.pite
12:26:06 PM PIT >> INFO : MINION : st.coverage.execute.CoverageMinion.main(CoverageMinion.java:84)

12:26:06 PM PIT >> SEVERE : Coverage generator Minion exited abnormally due to UNKNOWN_ERROR
Exception in thread "main" org.pitest.util.PitError: Coverage generation minion exited abnormally!

Please copy and paste the information and the complete stacktrace below when reporting an issue
VM : OpenJDK 64-Bit Server VM
Vendor : Oracle Corporation
Version : 12.0.1+12
Uptime : 37527
Input ->
 1 : -Dfile.encoding=UTF-8
 2 : -Duser.country=US
 3 : -Duser.language=en
 4 : -Duser.variant
BootClassPathSupported : false


Please copy and paste the information and the complete stacktrace below when reporting an issue
VM : OpenJDK 64-Bit Server VM
Vendor : Oracle Corporation
Version : 12.0.1+12
Uptime : 37528
Input ->
 1 : -Dfile.encoding=UTF-8
 2 : -Duser.country=US
 3 : -Duser.language=en
 4 : -Duser.variant
BootClassPathSupported : false

        at org.pitest.util.Unchecked.translateCheckedException(Unchecked.java:20)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:105)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:51)
        at org.pitest.mutationtest.tooling.MutationCoverage.runReport(MutationCoverage.java:115)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:121)
        at org.pitest.mutationtest.tooling.EntryPoint.execute(EntryPoint.java:51)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.runReport(MutationCoverageReport.java:87)
        at org.pitest.mutationtest.commandline.MutationCoverageReport.main(MutationCoverageReport.java:45)
Caused by: org.pitest.util.PitError: Coverage generation minion exited abnormally!

Please copy and paste the information and the complete stacktrace below when reporting an issue
VM : OpenJDK 64-Bit Server VM
Vendor : Oracle Corporation
Version : 12.0.1+12
Uptime : 37527
Input ->
 1 : -Dfile.encoding=UTF-8
 2 : -Duser.country=US
 3 : -Duser.language=en
 4 : -Duser.variant
BootClassPathSupported : false

        at org.pitest.coverage.execute.DefaultCoverageGenerator.gatherCoverageData(DefaultCoverageGenerator.java:148)
        at org.pitest.coverage.execute.DefaultCoverageGenerator.calculateCoverage(DefaultCoverageGenerator.java:89)
        ... 6 more

> Task :dcm:pitest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dcm:pitest'.
> Process 'command '/Library/Java/JavaVirtualMachines/openjdk-12.0.1.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

If you'd like to reproduce this failure:

```
$: g clone -b pitest https://github.com/vmware/declarative-cluster-management/tree/pitest
$: cd declarative-cluster-management
$: ./gradlew pitest
```

You can see the most recent commit to see the changes I introduced.

```
diff --git a/dcm/build.gradle b/dcm/build.gradle
index 082e424..dbc8aea 100644
--- a/dcm/build.gradle
+++ b/dcm/build.gradle
@@ -6,8 +6,10 @@
 plugins {
     id 'maven-publish'
     id 'signing'
+    id 'info.solidsoft.pitest' version '1.5.2'
 }

+
 dependencies {
     implementation "org.freemarker:freemarker:${freemarkerVersion}"
     implementation "org.apache.commons:commons-csv:${commonsCsvVersion}"
@@ -32,6 +34,12 @@ java {
     withSourcesJar()
 }

+pitest {
+    //adds dependency to org.pitest:pitest-junit5-plugin and sets "testPlugin" to "junit5"
+    junit5PluginVersion = '0.12'
+    // ...
+}
+
 publishing {
     repositories {
         maven {
```
Thanks!

--- a/pitest/src/main/java/org/pitest/classpath/ClassPath.java
+++ b/pitest/src/main/java/org/pitest/classpath/ClassPath.java
@@ -37,6 +37,8 @@ import org.pitest.util.ManifestUtils;
 import org.pitest.util.PitError;
 import org.pitest.util.StreamUtil;
 
+import static org.pitest.util.ManifestUtils.CLASSPATH_JAR_FILE_PREFIX;
+
 public class ClassPath {
 
   private static final Logger         LOG = Log.getLogger();
@@ -115,7 +117,7 @@ public class ClassPath {
 
   public static Collection<String> getClassPathElementsAsPaths() {
     final Set<String> filesAsString = new LinkedHashSet<>();
-    FCollection.mapTo(getClassPathElementsAsFiles(), file -> file.getPath(),
+    FCollection.mapTo(getClassPathElementsAsFiles(), File::getPath,
         filesAsString);
     return filesAsString;
   }
@@ -138,8 +140,8 @@ public class ClassPath {
    * @param elements existing elements
    */
   private static void addEntriesFromClasspathManifest(final Set<File> elements) {
-    Optional<File> maybeJar = elements.stream().filter( f -> f.getName().startsWith("classpath") && f.getName().endsWith(".jar"))
-    .findFirst();
+    Optional<File> maybeJar = elements.stream().filter(f -> f.getName().startsWith(CLASSPATH_JAR_FILE_PREFIX) && f.getName().endsWith(".jar"))
+        .findFirst();
     maybeJar.ifPresent(file -> elements.addAll(ManifestUtils.readClasspathManifest(file)));
   }
 

--- a/pitest/src/main/java/org/pitest/util/ManifestUtils.java
+++ b/pitest/src/main/java/org/pitest/util/ManifestUtils.java
@@ -34,6 +34,8 @@ import java.util.zip.ZipOutputStream;
  */
 public class ManifestUtils {
 
+  public static final String CLASSPATH_JAR_FILE_PREFIX = "pitest-classpath-jar-file-";
+
   // Method based on
   // https://github.com/JetBrains/intellij-community/blob/master/java/java-runtime/src/com/intellij/rt/execution/testFrameworks/ForkedByModuleSplitter.java
   // JetBrains copyright notice and licence retained above.
@@ -59,7 +61,7 @@ public class ManifestUtils {
     }
     attributes.put(Attributes.Name.CLASS_PATH, classpathForManifest.toString());
 
-    File jarFile = File.createTempFile("classpath", ".jar");
+    File jarFile = File.createTempFile(CLASSPATH_JAR_FILE_PREFIX, ".jar");
     try (BufferedOutputStream out = new BufferedOutputStream(new FileOutputStream(jarFile));
         ZipOutputStream jarPlugin = new JarOutputStream(out, manifest);
         )  {

